src/Symfony/Component/Filesystem/Tests/FilesystemTest.php

```diff
@@ -799,6 +799,80 @@ public function testSymlinkCreatesTargetDirectoryIfItDoesNotExist()
         $this->assertEquals($file, readlink($link2));
     }
 
+    public function testReadLink()
+    {
+        $this->markAsSkippedIfSymlinkIsMissing();
+
+        $file = $this->workspace.DIRECTORY_SEPARATOR.'file';
+        $link1 = $this->workspace.DIRECTORY_SEPARATOR.'dir'.DIRECTORY_SEPARATOR.'link';
+        $link2 = $this->workspace.DIRECTORY_SEPARATOR.'dir'.DIRECTORY_SEPARATOR.'subdir'.DIRECTORY_SEPARATOR.'link';
+
+        touch($file);
+
+        $this->filesystem->symlink($file, $link1);
+        $this->filesystem->symlink($link1, $link2);
+
+        $this->assertTrue(is_link($link1));
+        $this->assertEquals($file, $this->filesystem->readlink($link1));
+        $this->assertTrue(is_link($link2));
+        $this->assertEquals($link1, $this->filesystem->readlink($link2));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Filesystem\Exception\IOException
+     */
+    public function testReadLinkNotLink()
+    {
+        $file = $this->workspace.DIRECTORY_SEPARATOR.'file';
+        touch($file);
+        $this->filesystem->readlink($file);
+    }
+
+    /**
+     * @expectedException \Symfony\Component\Filesystem\Exception\IOException
+     */
+    public function testReadLinkFails()
+    {
+        $this->filesystem->readlink($this->workspace.DIRECTORY_SEPARATOR.'invalid');
+    }
+
+    public function testRealPath()
```

Comment:
That was a good idea, there are some inconsistency between Windows and Linux and between PHP and HHVM on that level. I'll fix that too.